worker/src/main/scala/com/lucidchart/piezo/Worker.scala

```diff
@@ -118,6 +124,7 @@ object Worker {
       override def run() {
         logger.info("received shutdown signal")
         runSemaphore.release()
+        shutdownSemaphore.acquire()
```

Comment:
@jjudd here's what happens:

1.) main starts and acquires a lock on the shutdownSemaphore (there's only one lock available, this could be a mutex instead if we wanted).

2.) main starts the scheduler and acquires a lock on the runSemaphore. I'm not 100% clear on the main loop here, but the effect is that main loops infinitely until runSemaphore is released.

3.) SIGTERM is sent by upstart (like when we restart the service). This immediately start the shutdown handler while _leaving main running_.

4.) the shutdown handler releases the runSemaphore, thereby getting main out of its infinite loop and starting the quartz shutdown process.

5.) the shutdown handler attempts to acquire shutdownSemaphore. It can't because main has it so it blocks until it can acquire.

6.) main finishes cleanup and calls shutdownSemaphore.release().

7.) the shutdown handler acquires shutdownSemaphore and finishes executing.

8.) the jvm is terminated.